### ISSUE TITLE
Inference on the custom camera 

### ISSUE THREAD

**Issue by sunwell1994** [zhyever/Monocular-Depth-Estimation-Toolbox#40]
Do we need to consider the camera intrinsic when performing inference with the pre-trained model? These models are trained with the known intrinsic matrics. I have not found the API to app the intrinsic for the inference. It may affect the estimation a lot.

**Comment by zhyever** [comment]
Yes, I agree that intrinsic parameters are crucial for accurate depth estimation. However, current methods do not consider it. Typically, they are trained on a specific dataset so that they can overfit the intrinsic parameters. In other words, the intrinsic parameters keep invariant during training and inference.

**Comment by sunwell1994** [comment]
> Yes, I agree that intrinsic parameters are crucial for accurate depth estimation. However, current methods do not consider it. Typically, they are trained on a specific dataset so that they can overfit the intrinsic parameters. In other words, the intrinsic parameters keep invariant during training and inference.


Thanks for your insight. It is really helpful.
I still have questions for intrinsic parameters.

1. Is there any method to normalize the intrinsic parameters smoothly so that the model can adapt to more cameras? For example, resize or crop can change intrinsic parameters. 
2. if the pre-trained model is trained from a specific dataset, is it reasonable to finetune the model on a custom dataset with different intrinsic parameters?

**Comment by zhyever** [comment]
1. I think that's related to a  practical domain adaptation problem. However, there is only limited research in this field. Previous depth DA(UDA) methods only utilize vKITTI and KITTI, where the camera intrinsic parameters are similar. I think this domain gap of camera intrinsic parameters is neglected. As for depth estimation, it seems that there are methods that try to embed the camera parameters into the model to achieve robust depth estimation training (a CVPR paper but not aim to solve the domain gap). I forget the paper title now. For DA(UDA), you can refer to my ECCV paper [STMono3D](https://github.com/zhyever/STMono3D), where we try to handle this issue in another manner.

2. Yes. It can bring a slight improvement when I conduct experiments. But how to improve the effectiveness of fine-tuning needs to be re-think since the discrepancy of camera parameters exists. I refer to MiDaS for some possible solutions to this issue.

**Comment by sunwell1994** [comment]
> 1. I think that's related to a  practical domain adaptation problem. However, there is only limited research in this field. Previous depth DA(UDA) methods only utilize vKITTI and KITTI, where the camera intrinsic parameters are similar. I think this domain gap of camera intrinsic parameters is neglected. As for depth estimation, it seems that there are methods that try to embed the camera parameters into the model to achieve robust depth estimation training (a CVPR paper but not aim to solve the domain gap). I forget the paper title now. For DA(UDA), you can refer to my ECCV paper [STMono3D](https://github.com/zhyever/STMono3D), where we try to handle this issue in another manner.
> 2. Yes. It can bring a slight improvement when I conduct experiments. But how to improve the effectiveness of fine-tuning needs to be re-think since the discrepancy of camera parameters exists. I refer to MiDaS for some possible solutions to this issue.

Thanks a lot.

**Comment by MyEyeAndEar** [comment]
good work
